plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -365,10 +365,12 @@ else if (type instanceof ScaledFloatType) {
             return new TypeAndDecoder(DOUBLE, new DoubleDecoder.Descriptor(path));
         }
         else if (type instanceof DateTimeType) {
-            if (((DateTimeType) type).getFormats().isEmpty()) {
-                return new TypeAndDecoder(TIMESTAMP_MILLIS, new TimestampDecoder.Descriptor(path));
+            List<String> formats = ((DateTimeType) type).getFormats();
+            String pattern = null;
+            if (!formats.isEmpty()) {
+                pattern = String.join("||", formats);
             }
-            // otherwise, skip -- we don't support custom formats, yet
+            return new TypeAndDecoder(TIMESTAMP_MILLIS, new TimestampDecoder.Descriptor(path, pattern));
```

Comment:
Change the constructor argument of `TimestampDecoder.Descriptor` to `List<String> patterns`.